questions/158/explanation.md

```diff
@@ -1,9 +1,10 @@
-Since C++11, `std::vector` has a one parameter constructor ( + allocator). §[vector.cons]¶3 in the standard):
-
-`explicit vector(size_type n, const Allocator& = Allocator())`
-
-which constructs a vector with `n` value-initialized elements. Each value-initialization calls the default `Foo` constructor, resulting in the output `aaaaa` .
-
-The "trick" is, that before C++11, `std::vector` had a 2 parameter constructor ( + allocator ), which constructed the container with `n` copies of the second parameter, which is defaulted to `T()`.
-
+Since C++11, `std::vector` has a one parameter constructor ( + optional allocator). §[vector.cons]¶3 in the standard:
+
+> `explicit vector(size_type n, const Allocator& = Allocator())`
+> Constructs a `vector` with `n` default-inserted elements using the specified allocator.
```

Comment:
```suggestion
Since C++11, `std::vector` has a one parameter constructor (+ optional allocator). §[vector.cons]¶4 in the standard:

> `constexpr explicit vector(size_type n, const Allocator& = Allocator());`
>
> *Effects*: Constructs a `vector` with `n` default-inserted elements using the specified allocator.
```